### DISCUSSION TITLE
Overlay in Split Windows

### DISCUSSION THREAD

**Discussion by karimlevallois** [kovidgoyal/kitty#7137] [Q&A]
HI all

I have the following startup layout:-

<img width="1680" alt="Screenshot 2024-02-17 at 18 56 46" src="https://github.com/kovidgoyal/kitty/assets/80129273/20dc16b4-b355-4f34-8a3b-8c21417ef65b">

When I use the following launch-actions.conf entry the file will open in the split that has focus, which I have set to be the main terminal screen (top-right). 

protocol file
mime text/*
action launch --type=overlay fish -c '$EDITOR $FILE_PATH'

<img width="1680" alt="Screenshot 2024-02-17 at 19 03 19" src="https://github.com/kovidgoyal/kitty/assets/80129273/c89e192f-5dcc-4719-b7d8-53c72fc39198">

However, ideally I want the opened file to launch in the large left-hand side split overlaying Neovim. I have tried multiple settings using the --location action, but nothing seems to work. Is this even possible?

Many thanks, Karim

**Comment by sbradnick** [comment]
I don't know if this is remotely related to your issue, but I learned the difficult lesson that for `action launch` items to behave as you'd *expect*, that you should put them in a file named `{$HOME}/.config/kitty/open-actions.conf`. You don't need to include that file (in `kitty.conf`) or anything like that, it'll get pulled in if it's present.

I spent an amount of time - I'm embarrassed to admit to - trying to `include launch-actions.conf` in my personalized config file and at best could only get a **new** neovim window with the $FILE_NAME item I'd referenced, never a new tab or the overlay `--type`; the second I took that exact same file and renamed it to `open-actions.conf` and removed the include -- it worked exactly as expected. I can probably blame the fact that I was trying to zip through the intro vid and not listening to it as the reason (completely my fault). And kitty was trying to tell me I was doing it wrong by printing messages akin to 'I don't know what "action" or "launch" is' (or they're invalid) and I'd ignored that vs. understanding what it was really telling me.

Now I can use the `--hyperlink` functionality of ripgrep and ls to jump right into specific lines of files and a fun little overlay for images via the `icat` kitten. Cool stuff :+1:

**Comment by kovidgoyal** [comment]
An overlay window always appears over the active window. --location is
ignored. If you want to do some more sophisticated placement use the
remote control facility with a script.

https://sw.kovidgoyal.net/kitty/actions/#action-remote_control_script

In your script use
kitty @ ls to get details about all windows, extract the id of the
window you want to overlay, make that the active window with kitty @
set-focus and then use kitty @ launch to actually launch the overlay.
After launching the overlay you can move focus back to the original
window, if so desired.

**Reply by karimlevallois** [reply]
Many thanks for such a detailed response, much appreciated.

Can this be done within the launch command itself? It will always be KITTY_WINDOW_ID 1 with this startup.

I think I've got it working with the following: -

```
protocol file
mime text/*
action first_window
action launch --type=overlay fish -c '$EDITOR $FILE_PATH'
```
